packages/feature-detector/scripts/patch-type-definition.ts

```diff
@@ -0,0 +1,106 @@
+// assumes being ran in packages/feature-detector
+
+import { get } from 'node:https'
+import { writeFile } from 'node:fs/promises'
+import { join } from 'node:path'
+
+const repositoryBaseUrl =
+  'https://raw.githubusercontent.com/microsoft/TypeScript/main/lib'
+const destinationFile = join('src', 'utils/type-definition.txt')
+
+async function writeTypeDefinition() {
+  const aggregatedContent = await fetchTypeDefinitions([
+    'lib.es2022.d.ts',
+    'lib.dom.d.ts',
+  ])
+  await writeFile(destinationFile, patchContent(aggregatedContent).join('\n'))
+}
+
+async function fetchTypeDefinitions(files: string[]) {
+  const result = []
+  for (const file of files) {
+    result.push(...(await fetchTypeDefinition(file)))
+  }
+  return result
+}
+
+async function fetchTypeDefinition(file: string): Promise<string[]> {
+  // I wish we could use node@18's fetch...
+  const content = await new Promise<string>((resolve, reject) =>
+    get(`${repositoryBaseUrl}/${file}`, (response) => {
+      let content = ''
+      response.setEncoding('utf8')
+      response.on('data', (chunk: string) => (content += chunk))
```

Comment:
could be nice to extract into a function, similar to this:
https://github.com/feross/simple-concat/blob/master/index.js

Reply:
see my comment above: this is just a build script, that hopefully will be gone when typescript will support `Response.json()`, or that can use undici if we always build on node18 as explained.